### PR TITLE
Fix namespace resolution issue in node copy and undo

### PR DESCRIPTION
**Pull request by ke-yu**
### Purpose

This PR fixes defect [MAGN-7917 Resetting preset causes CBN to show "Dereferencing..." error due to namespace collision](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7917). Namespace collision not only happens in resetting preset, but also happens in undo and copy/paste. 

A .dyn file's `Workspace` XML element may contains a child element `NamespaceResolutionMap` which contains a key-value pair list to map partial names to fully qualified names, for example, 
```
<NamespaceResolutionMap>
    <ClassMap partialName="Color" resolvedName="DSCore.Color" assemblyName="DSCoreNodes.dll" />
</NamespaceResolutionMap>
```
and `ElementResolver` will use this map to resolve partial names. Currently there are two kinds of node will use `ElementResolver`: code block node and input node.

Ideally, all code block nodes or input nodes should use global `ElementResolver`, which is from current workspace, to do namespace resolution. But in the process of .dyn file loading, nodes are created *before* the creation of workspace (which will read `NamespaceResolutionMap` and create `ElementResolver`), so to properly parse code in code block node or input node, previously code block node and input node will try to get `NamespaceResolutionMap` directly from root XML document and create an instance of `ElementResolver` for its own use:
```
protected override void DeserializeCore(XmlElement nodeElement, SaveContext context) 
{
    ....
    // CodeBlockUtils.DeserializeElementResolver() will parse the root XML element
    var resolutionMap = CodeBlockUtils.DeserializeElementResolver(nodeElement);
    ElementResolver = new ElementResolver(resolutionMap);
}

public static IDictionary<string, KeyValuePair<string, string>> DeserializeElementResolver( XmlElement nodeElement)
{
    var xmlDoc = nodeElement.OwnerDocument;
    ...
    var nodes = xmlDoc.GetElementsByTagName("NamespaceResolutionMap");
    ...
}
```
And after node is loaded, for the consequent code parsing it will use `ElementResolver` from workspace. But it doesn't work for undo/redo and preset because for these two cases only the XML data of node itself is serialized and in deserialization the root XML document doesn't contain other elements, so code block node doesn't have a valid `ElementResolver` to resolve partial namespace.

In this PR, we try to create `ElementResolver` *before* nodes are loaded, and pass `ElementResolver` to the corresponding nodes. 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@aparajit-pratap

### CODE REVIEW

File: src/DynamoCore/Models/NodeLoaders/ZeroTouchNodeLoader.cs [hunk 1]
```diff
@@ -22,7 +23,7 @@ public ZeroTouchNodeLoader(LibraryServices libraryServices)
             this.libraryServices = libraryServices;
         }
 
-        public NodeModel CreateNodeFromXml(XmlElement nodeElement, SaveContext context)
+        public NodeModel CreateNodeFromXml(XmlElement nodeElement, SaveContext context, ElementResolver resolver)
```
Comment (aparajit-pratap) [inline src/DynamoCore/Models/NodeLoaders/ZeroTouchNodeLoader.cs]: Why do zero touch nodes need an `ElementResolver`? It should be applicable only to CBN's and input nodes, right?

**Comment by aparajit-pratap** [comment]
Thanks @ke-yu for fixing this! LGTM!

**Comment by ke-yu** [comment]
@aparajit-pratap , as `ApplyPreset` command doesn't work in recorded test, I don't add test case for it. (The reason is `ApplyPreset` command needs to know which workspace it applies to, so it has a parameter for workspace guid. But recorded test frame doesn't save workspace id  to xml file, so for each run workspace will have different GUID, and `ApplyPreset` command will fail in recorded test).

**Comment by aparajit-pratap** [comment]
LGTM!

**Pull request merged**